### PR TITLE
Fix typos and replace 'codespell' with 'typos'

### PR DESCRIPTION
**Pull request by jayvdb**
Uses https://github.com/taiki-e/install-action to install https://github.com/crate-ci/typos in CI

This finds many more/different typos than https://github.com/codespell-project/codespell , while having very few false positives.

### CODE REVIEW

File: .config/typos.toml
Comment (jayvdb) [inline .config/typos.toml]: this is in an `extern` variable type, so I didnt want to rename it without discussion.

However, especially given valkey is still unstable, maybe this is a good time to do the rename.

Does valkey want to retain API or ABI compatibility?

Comment (jayvdb) [reply .config/typos.toml]: I see https://github.com/valkey-io/valkey/issues/43

Comment (jayvdb) [reply .config/typos.toml]: Note this isnt listed in the codespell wordlist.

File: src/server.c
Comment (jayvdb) [inline src/server.c:2578]: https://en.wiktionary.org/wiki/cancelability and the note at the bottom of https://www.merriam-webster.com/dictionary/cancel suggests this was an acceptable alternative , but given it only appears once, and it looks like it will get flagged by multiple spellcheckers, it seems wise to just switch to using the spelling commonly used.

**Comment by szepeviktor** [comment]
Please consider dropping codespell.
Its dictionary is included in `typos`: https://github.com/crate-ci/typos/tree/master/crates/codespell-dict

**Comment by PingXie** [comment]
> Please consider dropping codespell. Its dictionary is included in typos: https://github.com/crate-ci/typos/tree/master/crates/codespell-dict

I am not familiar with codespell nor spellchecker in general. Are you saying that the existing spellchecker is sufficient?

BTW, @jayvdb, all other changes LGTM. Thanks!

**Comment by jayvdb** [comment]
I did a bit of digging and can confirm `typos` is a superset of `codespell` wrt to misspelling detection capability, and finer grained configuration.  `codespell` has a few UX features that `typos` doesnt have, such as an interactive TUI mode.  Anyone wanting those features can continue to use `codespell` locally.

**Review by zuiderkwast — COMMENTED** [review]
LGTM

How can I run typos locally? It's useful to be able to run it before pushing, instead of waiting for CI failure to find out.

**Comment by jayvdb** [comment]
> LGTM
> 
> How can I run typos locally? It's useful to be able to run it before pushing, instead of waiting for CI failure to find out.

See https://github.com/crate-ci/typos?tab=readme-ov-file#install

**Comment by szepeviktor** [comment]
> How can I run typos locally?

Just download typos, it is a single binary. Nicely fits in `~/bin/`
https://github.com/crate-ci/typos/releases

**Review by zuiderkwast — APPROVED** [review]
Sounds good then.

I'm leaving this open for a little while to let others look before merging.

File: .config/typos.toml
Comment (madolson) [inline .config/typos.toml]: I would like to document why these two files are excluded:
```suggestion
    # crc16_slottable includes a bunch of pre-generated random strings.
    "src/crc16_slottable.h",
    # I don't know why eval.c is excluded, so would be good to put something here.
    "src/eval.c",
```

Comment (szepeviktor) [reply .config/typos.toml]: It is vendored code.
https://github.com/valkey-io/valkey/blob/af1b0de92de2ace4a01e74cf9ed7d7d6f3dedf70/src/eval.c#L2

Comment (madolson) [reply .config/typos.toml]: Well, that's more of a statement that our copyrights aren't accurate. The code was heavily modified in the last few years while we worked on it in Redis. https://github.com/valkey-io/valkey/commits/unstable/src/eval.c It was modified within the last 3 weeks.

Comment (jayvdb) [reply .config/typos.toml]: I've added the regex exception needed to allow `eval.c` to be checked; and added a comment on `crc16_slottable.h`

**Comment by PingXie** [comment]
Looks like we are in general agreement with the bulk of the changes. I am going to merge this PR now to reduce the PR backlog. We can always take incremental changes.

**Pull request merged**
